core/citrus-api/src/main/java/com/consol/citrus/variable/SegmentVariableExtractor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2006-2010 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.consol.citrus.variable;
+
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.exceptions.CitrusRuntimeException;
+import com.consol.citrus.spi.ResourcePathTypeResolver;
+import com.consol.citrus.spi.TypeResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Class extracting values of segments of VariableExpressions.
+ *
+ * @author Thorsten Schlathoelter
+ */
+public interface SegmentVariableExtractor {
+
+    /** Logger */
+    Logger LOG = LoggerFactory.getLogger(SegmentVariableExtractor.class);
```

Comment:
write in lowercase letters, or is it commonly written in uppercase in this module? (there are some other `Logger` instaces as well in this changeset..)

Reply:
yes loggers usage needs to be unified in the whole framework. I think they all should be `private static final`. And by convention static final members are spelled in uppercase. Would be happy to have this addressed in a separate PR

Reply:
I have added an issue. We can discuss the final definition for the log there.
https://github.com/citrusframework/citrus/issues/829